mycroft/client/speech/mic.py

```diff
@@ -513,32 +520,24 @@ def _compile_metadata(self):
         else:
             model_hash = '0'
 
-        return {
-            'name': self.wake_word_name.replace(' ', '-'),
-            'engine': md5(ww_module.encode('utf-8')).hexdigest(),
-            'time': str(int(1000 * get_time())),
-            'sessionId': SessionManager.get().session_id,
-            'accountId': self.account_id,
-            'model': str(model_hash)
-        }
+        return dict(
+            name=self.wake_word_name.replace(' ', '-'),
+            engine=md5(ww_module.encode('utf-8')).hexdigest(),
+            engine_name=ww_module.lower().strip('hotword'),
+            time=str(int(1000 * get_time())),
+            sessionId=SessionManager.get().session_id,
+            accountId=self.account_id,
+            model=str(model_hash)
+        )
 
     def trigger_listen(self):
         """Externally trigger listening."""
         LOG.debug('Listen triggered from external source.')
         self._listen_triggered = True
 
     def _upload_wakeword(self, audio, metadata):
-        """Upload the wakeword in a background thread."""
-        LOG.debug(
-            "Wakeword uploading has been disabled. The API endpoint used in "
-            "Mycroft-core v20.2 and below has been deprecated. To contribute "
-            "new wakeword samples please upgrade to v20.8 or above."
-        )
-        # def upload(audio, metadata):
-        #     requests.post(self.upload_url,
-        #                   files={'audio': BytesIO(audio.get_wav_data()),
-        #                          'metadata': StringIO(json.dumps(metadata))})
-        # Thread(target=upload, daemon=True, args=(audio, metadata)).start()
+        """Upload the wake word to the backend servers."""
+        self.device_api.upload_wake_word(audio, metadata)
```

Comment:
This was previously done by launching a background thread to not block. Should that be used now as well?